dlt/destinations/athena/athena.py

```diff
@@ -0,0 +1,262 @@
+from dlt.destinations.typing import DBApiCursor
+from dlt.common.destination.reference import LoadJob
+from dlt.common.destination.reference import TLoadJobState
+from dlt.common.storages import FileStorage
+from dlt.common.schema.typing import VERSION_TABLE_NAME
+from dlt.common.schema.utils import add_missing_hints
+
+from dlt.destinations.job_client_impl import SqlJobClientBase, StorageSchemaInfo
+from dlt.destinations.athena.configuration import AthenaClientConfiguration
+from dlt.destinations.job_impl import NewReferenceJob
+from dlt.destinations import path_utils
+
+SCT_TO_HIVET: Dict[TDataType, str] = {
+    "complex": "string",
+    "text": "string",
+    "double": "double",
+    "bool": "boolean",
+    "date": "date",
+    "timestamp": "timestamp",
+    "bigint": "bigint",
+    "binary": "binary",
+    "decimal": "decimal(%i,%i)"
+}
+
+HIVET_TO_SCT: Dict[str, TDataType] = {
+    "varchar": "text",
+    "double": "double",
+    "boolean": "bool",
+    "date": "date",
+    "timestamp": "timestamp",
+    "bigint": "bigint",
+    "binary": "binary",
+    "decimal": "decimal"
+}
+
+# add a formatter for pendulum to be used by pyathen dbapi
+def _format_pendulum_datetime(formatter: Formatter, escaper: Callable[[str], str], val: Any) -> Any:
+    # copied from https://github.com/laughingman7743/PyAthena/blob/f4b21a0b0f501f5c3504698e25081f491a541d4e/pyathena/formatter.py#L114
+    val_string = val.strftime("%Y-%m-%d %H:%M:%S.%f")
+    return f"""TIMESTAMP '{val_string}'"""
+
+class DLTAthenaFormatter(DefaultParameterFormatter):
+    def __init__(self) -> None:
+        formatters = deepcopy(_DEFAULT_FORMATTERS)
+        formatters[DateTime] = _format_pendulum_datetime
+        super(DefaultParameterFormatter, self).__init__(
+            mappings=formatters, default=None
+        )
+
+class DoNothingJob(LoadJob):
+    """The most lazy class of dlt"""
+
+    def __init__(self, file_path: str) -> None:
+        super().__init__(FileStorage.get_file_name_from_file_path(file_path))
+
+    def state(self) -> TLoadJobState:
+        # this job is always done
+        return "completed"
+
+    def exception(self) -> str:
+        # this part of code should be never reached
+        raise NotImplementedError()
+
+class AthenaSQLClient(SqlClientBase[Connection]):
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, dataset_name: str, config: AthenaClientConfiguration) -> None:
+        super().__init__(None, dataset_name)
+        self._conn: Connection = None
+        self.config = config
+
+    @raise_open_connection_error
+    def open_connection(self) -> Connection:
+        self._conn = connect(
+            schema_name=self.dataset_name,
+            s3_staging_dir=self.config.query_result_bucket,
+            region_name=self.config.credentials.aws_region,
+            aws_access_key_id=self.config.credentials.aws_access_key_id,
```

Comment:
Doesn't this force the use of aws access keys in secrets.toml or the respective env vars? It won't be able to fall back onto other boto3 auth methods.

Reply:
Something like this
```
@raise_open_connection_error
    def open_connection(self) -> Connection:
        self._conn = connect(
            schema_name=self.dataset_name,
            s3_staging_dir=self.config.query_result_bucket,
            **self.config.credentials.to_native_representation())
        return self._conn
```


but also requires we modify the aws_credentials spec function that is used to match the region name arg. From common.specs.aws_credentials.py

```
def to_native_representation(self) -> Dict[str, Optional[str]]:
        """Return a dict that can be passed as kwargs to boto3 session"""
        d = dict(self)
        d['profile_name'] = d.pop('aws_profile')  # boto3 argument doesn't match env var name
        d['region_name'] = d.pop('aws_region')  # boto3 argument doesn't match env var name
        return d
```


Or we can expose the session from the aws spec?

Reply:
good point! look here https://github.com/dlt-hub/dlt/pull/534/files#diff-86471f800e38d78e3737c979d879eb02b8a0ca3e956068747989bba466503d68
I think it is doing exactly what you describe, you can also pass boto3 instead of credentials dict
@sh-rp `to_native_representation` should return dict with names compatible with Athena, let's try to pass it unstructured as @lukas-gust shows